AEPServices/Sources/ui/fullscreen/FullscreenMessage+FrameCalculation.swift

```diff
@@ -0,0 +1,171 @@
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+import UIKit
+
+extension FullscreenMessage {
+    // MARK: - internal vars
+
+    /// calculated frames are used for animation
+    /// frameWhenVisible is the frame when the message is visible to the user
+    var frameWhenVisible: CGRect {
+        return CGRect(x: originX, y: originY, width: width, height: height)
+    }
+
+    /// frameBeforeShow considers displayAnimation and positions the message appropriately
+    var frameBeforeShow: CGRect {
+        guard let displayAnimation = settings?.displayAnimation else {
+            return frameWhenVisible
+        }
+
+        switch displayAnimation {
+        case .top:
+            return CGRect(x: originX, y: -(height + originY), width: width, height: height)
+        case .bottom:
+            return CGRect(x: originX, y: screenHeight, width: width, height: height)
+        case .right:
+            return CGRect(x: screenWidth, y: originY, width: width, height: height)
+        case .left:
+            return CGRect(x: -(screenWidth + originX), y: originY, width: width, height: height)
+        case .center:
+            return CGRect(x: screenWidth * 0.5, y: screenHeight * 0.5, width: 0, height: 0)
+        default:
+            return frameWhenVisible
+        }
+    }
+
+    /// frameAfterDismiss considers dismissAnimation and positions the message appropriately
+    var frameAfterDismiss: CGRect {
+        guard let dismissAnimation = settings?.dismissAnimation else {
+            return frameWhenVisible
+        }
+
+        switch dismissAnimation {
+        case .top:
+            return CGRect(x: originX, y: -(height + originY), width: width, height: height)
+        case .bottom:
+            return CGRect(x: originX, y: screenHeight, width: width, height: height)
+        case .right:
+            return CGRect(x: screenWidth, y: originY, width: width, height: height)
+        case .left:
+            return CGRect(x: -(screenWidth + originX), y: originY, width: width, height: height)
+        case .center:
+            return CGRect(x: screenWidth * 0.5, y: screenHeight * 0.5, width: 0, height: 0)
+        default:
+            return frameWhenVisible
+        }
+    }
+
+    // returns the width of the screen, measured in points
+    var screenWidth: CGFloat {
+        return UIScreen.main.bounds.width
+    }
+
+    var screenHeight: CGFloat {
+        return UIScreen.main.bounds.height
+    }
+
+    // MARK: - private vars
+
+    // width in settings represents a percentage of the screen.
+    // e.g. - 80 = 80% of the screen width
+    // default value is full screen width
```

Comment:
should these be doc comments?

Reply:
additionally, for ease of reading i think periods can be added to the end of the sentences which are missing them.